src/cartesian/Area.js

```diff
@@ -198,9 +200,9 @@ class Area extends Component {
   state = { isAnimationFinished: true };
 
   componentWillReceiveProps(nextProps) {
-    const { animationId, points, baseLine } = this.props;
+    const { restartAnimationOnChange, animationId, points, baseLine } = this.props;
 
-    if (nextProps.animationId !== animationId) {
+    if (!restartAnimationOnChange && nextProps.animationId !== animationId) {
```

Comment:
@delold  the default value of `restartAnimationOnChange`  is set to be `false` now. 

Then if `nextProps.animationId !== animationId`, `prevPoints` will be updated. Then if `!_.isEqual(prevPoints, points)` , animation will be triggered.

if the value of `restartAnimationOnChange` is set to be `true`,   `prevPoints` will always be `null`, then every time the component will be rendered with animation.

is this behavior what you want?

Reply:
I believe this is not the case, as the condition basically prevents calling `cachePrevData` with previous data. 

When the `restartAnimationOnChange` is set to `false`, the `cachePrevData` should either be a no-op or be called with an empty array. 

PS: forgive my deleted comments, it is quite late here and I'm not thinking that straight 😅

Reply:
It is true though, that this can be considered confusing, it took me a while to understand it. If you want, I can redo the condition.